企業情報一覧ページを作成する

Jyoumama opened the pull request:
## Issue

- #25 

## 概要
🔹 PRの目的
このプルリクエストでは、企業情報一覧ページの作成と、それに伴うコードの改善を行いました。

🔹 主な変更点
1.企業情報一覧ページの作成
- ログインしたユーザーのみが企業情報一覧ページに遷移できるように実装。
- `fetchCompanies` 関数のエラーハンドリングを強化。
- 企業情報一覧のテーブルスタイリングを改善。
2. Supabase関連の更新
- 最新の `database.types.ts` を適用し、型定義を更新。
- Supabase クライアントの設定を見直し、適切にデータ取得できるよう調整。
3. 不要なファイルの整理
`pjord_svg.code-search` という作業途中で意図せず作成してしまった検索関連ファイルを削除。
- 一時的な `database.types.new.ts` を削除し、正式な型定義を `database.types.ts` に反映。

🔹 変更の確認方法
1.`feature/create-companies-list` ブランチをローカルにチェックアウト：
```bash
git checkout feature/create-companies-list
```
2. アプリを起動：
```bash
npm run dev
```
3. 確認ポイント
- ログイン後、企業情報一覧ページ (/companies) へ正しく遷移できるか？
- 企業情報のテーブルが適切に表示され、スタイリングが適用されているか？

📌 実施した主なコミット一覧

✅ 企業情報一覧ページの作成
- コミット: ceaeae1
 変更点:
- 企業情報一覧ページ (`Companies.tsx`) のレイアウトを作成。
- ログインユーザーのみアクセスできるよう、認証チェックを追加。
✅ 不要な検索関連ファイルの削除
- コミット: 7050aee
変更点:
- `pjord_svg.code-search` という未使用のファイルを削除。
✅ Supabaseの型定義とクライアント設定を更新
- コミット: edb8a1d
変更点:
- `database.types.ts` を `npx supabase gen types` で最新化。
- `supabase.ts` の設定を調整し、型エラーを防ぐ対応を実施。
✅ 一時的な database.types.new.ts の削除
- コミット: 77d6a30
変更点:
`database.types.ts` への適用後、一時ファイルを削除。

## 変更前
- 変更前（企業情報一覧のリンクなし）
<img width="1328" alt="スクリーンショット 2025-01-27 20 16 51" src="https://github.com/user-attachments/assets/76f2f960-1008-45ae-b6c7-a30892db66bf" />

## 変更後
- ログイン認証を実施（agentのローカルで使用するユーザー情報）
<img width="1317" alt="スクリーンショット 2025-01-27 20 17 35" src="https://github.com/user-attachments/assets/04f20503-2ccf-47ff-8391-cee6b2c5ac93" />

- ログインしたら企業情報一覧のリンクが表示される
<img width="1321" alt="スクリーンショット 2025-01-27 20 17 53" src="https://github.com/user-attachments/assets/67d0fb08-8500-4113-8df1-ab99f60595c1" />

- リンクをクリックしたら企業情報一覧ページに飛ぶ(※編集・削除ボタンはdefaultでありますが、別Issueのようなのでまだリンク等は出来ていません。）
<img width="1439" alt="スクリーンショット 2025-01-28 16 25 44" src="https://github.com/user-attachments/assets/f819221e-1a95-49e9-b1f9-4138d9429cb4" />

Jyoumama committed:
ログイン機能とログインユーザーのみが企業情報一覧のリンク画面と企業情報一覧ページに遷移する機能を実装

Jyoumama committed:
fetchCompanies関数にエラーハンドリングを追加しテーブルにボーダーとパディングを追加してスタイリングを改善

Jyoumama committed:
企業ページのレイアウトとコンポーネントを実装・調整

Jyoumama committed:
不要な検索関連ファイルを削除

Jyoumama committed:
機能追加: Supabaseの型定義とクライアント設定を更新

Jyoumama committed:
一時的な database.types.new.ts ファイルを削除

Jyoumama commented:
@kazuma-naka さん、
お疲れ様です〜。お手隙の際に、レビューをお願いします。🙏

kazuma-naka commented:
@Jyoumama さん、

お疲れ様です。
手元で起動してみました。
`companies` の一覧が表示されませんでした。

supabase を起動時に [seed.sql](https://github.com/fjordllc/agent/blob/main/supabase/seed.sql)  を読み込むと以下の SQL が実行されるという認識を私は持っています。 @Jyoumama さんのスクショのデータが `seed.sql` と異なって見えるのですがどのようにデータを追加されたのか気になりました。

[日報](https://bootcamp.fjord.jp/reports/107599)を拝読しまして Companies の一覧が表示されない理由は RLS が関係していますでしょうか？

```bash
INSERT INTO companies (name, website, memo) VALUES ('株式会社ロッカ', 'https://lokka.jp', '良い会社');
INSERT INTO companies (name, website, memo) VALUES ('株式会社リンゴ', 'https://example.com/apple', 'リンゴは良い会社');
INSERT INTO companies (name, website, memo) VALUES ('株式会社バナナ', 'https://example.com/banana', 'バナナは良い会社');
INSERT INTO companies (name, website, memo) VALUES ('株式会社オレンジ', 'https://example.com/orange', 'オレンジは良い会社');
```

![companies_table](https://github.com/user-attachments/assets/1cb5c14c-57c7-48e1-8cf2-9238c2c39da4)

![supabase_table_companies](https://github.com/user-attachments/assets/765279d1-c49e-4c0f-8424-3ed1ae31f042)

Jyoumama commented:
@kazumaNさん、確認＆コメントをいただいてありがとうございます！
日報も確認いただいてありがとうございます！

とり急ぎ、返信いたします。結論、これからこちらのプルリクエストにkazumaNさんが適用されている`seed.sql`をデータごと
以下のコマンドで取り込んでコミット&プッシュする作業に取り掛かりたいと考えています！

```bash
supabase db reset
supabase db seed
```
私が入力した企業のダミーデータがkazumaNさんのローカル環境でご覧になれない理由は、私が`seed.sql`を更新せず、プルリクエストに含めていないためです。
現在ご確認いただいているこの作業ブランチを`main`から切る際に、`main`を最新の状態にしてからローカルに`pull`して、作業ブランチを作っていますが、既に`seed.sql`に確認用のダミーデータが記載されているということを私が認識しておらず、自分のsupabaseのデータベースにスクショになるテーブルデータを入力して、表示の確認をしておりました！そのデータがRLSの設定が誤っていたため、私のローカル環境でアプリに表示されない状態だったのですが、RLSの設定を修正し、表示される様になったため、そのまま`seed.sql`をプルリクエストに含めずにレビュー依頼をしてしまっておりました〜。
修正しましたら、改めてレビュー依頼をさせていただけますか？
ご連絡は、Discordにもした方が良いでしょうか？
お忙しいところ、恐縮です。よろしくお願いします！☺️

kazuma-naka commented:
> @kazumanさん、確認＆コメントをいただいてありがとうございます！ 日報も確認いただいてありがとうございます！
> 
> とり急ぎ、返信いたします。結論、これからこちらのプルリクエストにkazumaNさんが適用されている`seed.sql`をデータごと 以下のコマンドで取り込んでコミット&プッシュする作業に取り掛かりたいと考えています！
> 
> ```shell
> supabase db reset
> supabase db seed
> ```
> 
> 私が入力した企業のダミーデータがkazumaNさんのローカル環境でご覧になれない理由は、私が`seed.sql`を更新せず、プルリクエストに含めていないためです。 現在ご確認いただいているこの作業ブランチを`main`から切る際に、`main`を最新の状態にしてからローカルに`pull`して、作業ブランチを作っていますが、既に`seed.sql`に確認用のダミーデータが記載されているということを私が認識しておらず、自分のsupabaseのデータベースにスクショの内容のテーブルデータを入力して、表示の確認をしておりました！そのデータがRLSの設定が誤っていたため、私のローカル環境でアプリに表示されない状態だったのですが、RLSの設定を修正し、表示される様になったため、そのまま`seed.sql`をプルリクエストに含めずにレビュー依頼をしてしまっておりました〜。 修正しましたら、改めてレビュー依頼をさせていただけますか？ ご連絡は、Discordにもした方が良いでしょうか？ お忙しいところ、恐縮です。よろしくお願いします！☺️

連絡はこちらで大丈夫ですよ！

Jyoumama committed:
企業一覧の更新 & シードデータ修正 & RLSポリシー追加

kazuma-naka commented on supabase/migrations/20250129044022_add_public_read_access.sql:
この POLICY ですと認証されたユーザー、 未認証のユーザー関係なくすべてのユーザーが companies テーブルにアクセスできると思います。
会社一覧を参照する際にログインが必要でしたので気になりました。
どのユーザーが companies テーブルに何の操作できるのか、定義してあげる必要があると感じました。

kazuma-naka replied:
https://supabase.com/docs/guides/database/postgres/roles

こちらに supabase の roles の一覧があります

Jyoumama committed:
chore: SupabaseのRLS設定を修正

Jyoumama committed:
fix: Supabaseクライアントの設定を修正

Jyoumama committed:
feat: 企業一覧を認証チェック付きで実装

Jyoumama committed:
chore: TypeScriptのパス設定を更新

Jyoumama committed:
chore: 依存関係を更新

Jyoumama committed:
refactor: RLSの設定を統一し、ファイル名を変更

Jyoumama committed:
refactor:余計な記号などを削除

Jyoumama committed:
refactor:余計な記号などを削除

Jyoumama committed:
不要なマイグレーションファイルadd_public_read_access.sqlを削除しファイル名をTimestampで正しく取得

Jyoumama committed:
supabase.tsのデバッグログを削除

Jyoumama commented:
@kazuma-naka さん、
お忙しいところすみません！
企業一覧の認証チェック付きの実装や、RLSの統一を行いました。
以下の変更点について、レビューをお願いいたします。

🛠 主な変更点
1. 企業一覧を認証チェック付きで実装 (Companies.tsx)
- 認証されていないユーザーはデータを取得できないようにしました。
- Supabase の auth.getUser() を使い、ログインユーザーのみがデータを取得できる仕様にしました。
2. RLS (Row-Level Security) の統一 (supabase/migrations)
- `20250129044022_add_public_read_access.sql` の記述を `select_companies_policy.sql`に統一しました。
- authenticated ユーザーに SELECT 権限を付与しました。
3. 細かいリファクタリング
- `supabase.ts`, `tsconfig.json` などの不要な記号や余計な記述を整理しました。
- スタイル調整 (テーブルのボーダーを統一) も行いました。

👀 レビューしてほしい点
-認証周りの実装 (fetchCompanies() の処理)
-Supabase の RLS 設定の統一 (select_companies_policy.sql)
-その他、改善点があれば教えてください！

お手すきの際に、レビューよろしくお願いします！🙏

kazuma-naka commented on src/components/Companies.tsx:
Test は別ファイルに記述するとよいと思います

Jyoumama replied:
Companies.test.tsxを作成し、テストは別に記述しました。
やっとパスしました〜。

kazuma-naka replied:
`handleLogin()` を使って `init()`されています。
ローカルのテストユーザーでログインをされているのですが必要がないと思うのですがいかがでしょうか？

kazuma-naka commented on package.json:
こちらの local と remote はなぜ分かれているのでしょうか？

Jyoumama replied:
本番環境の動作を確認する必要があるかなあ、と考えたので、自分でsupabaseのクラウド環境にリモートのagentのプロジェクトを（カリキュラムで実施したように）作成しており、ターミナルでremoteとlocalを別々に起動できるように
設定しています。よって、`package.json`にそのコマンドを設定しています。
なので、package.jsonは当初プルリクエストには含めるべきではないなあ、と考えていたのですが、今回含めたのは、この動作確認の必要性を確認したかったからです。ですので、駒形さんなどに確認し、無用！であれば、このpackage.jsonはまーじすべきではないので、自分の環境のみで使用したいと考えます。

また、想定remoteの方の環境変数は、当然、Git管理していませんので、（.gitignoreするようにしている）レビュアーには見ることができませんが、自分のPCのプロジェクト上では`.env.remote`と`env.local`として別管理しています。

# supabaseクラウド環境（リモートの Supabase）←本番環境（想定）で動作確認
NEXT_PUBLIC_SUPABASE_URL=https://省略
NEXT_PUBLIC_SUPABASE_ANON_KEY=・・・・

# ローカル環境（self-hosted）←こちらでチーム開発
NEXT_PUBLIC_SUPABASE_URL=http://127.0.0.1:54321
NEXT_PUBLIC_SUPABASE_ANON_KEY=eyJhbGciOiJI...

上記の理由になります。混乱させてしまい、申し訳ありません。。。

kazuma-naka replied:
@supabase/postgrest-js と npm-run-all は必要ないと思うのですがいかがでしょうか？

kazuma-naka replied:
すいません。 npm-run-all は必要かもしれません。

Jyoumama commented on src/lib/supabase.ts:
`package.json`のlocalとremoteがなぜ、分かれているのか？のコメント欄で記載したことと関連していますが、
自分のPC上では、`env.local`と`env.remote`で環境変数を分けています。
よって、こちらhttp://127.0.0.1:54321/をわざわざ設定するようにしていました。
おっしゃる通り、環境変数が常に設定される運用なら、デフォルトの "http://127.0.0.1:54321" は不要で、
代わりに、環境変数が未設定ならエラーを投げるようにするなど、した方が良いでしょうか？？

Jyoumama committed:
Supabaseのマイグレーションを更新

- create_companies.sqlをmainの内容に再修正し、スキーマ変更を適用
- 不要になったポリシー関連のマイグレーションファイルを削除
- データベースのスキーマの不整合を修正

Jyoumama committed:
Jestの設定とセットアップを更新

Jyoumama committed:
Supabase のモックを修正（認証とデータ取得の処理

Jyoumama committed:
Companies コンポーネントの認証処理を改善

Jyoumama committed:
ログイン・サインアップページの認証フローを修正

Jyoumama committed:
依存関係のファイルを更新：.env.locaolとremoteに分けていた仕様を元に戻した

Jyoumama committed:
Companies コンポーネントのテストを追加

Jyoumama committed:
コンフリクト解消時に発生した依存関係ファイルの更新の抜けを修正

Jyoumama committed:
依存関係の更新：@supabase/supabase-jsのバージョンを2.47.3に合わせる

Jyoumama commented:
お手隙の際にご確認をお願いします。🙏

今回は、以下の変更を行いました：

### 主な変更点
1. **mainブランチの統合**
   - mainブランチを feature/create-companies-list に統合し、以下のファイルで発生したコンフリクトを解消しました：
     - `package-lock.json`
     - `src/lib/database.types.ts`

2. **機能修正**
   - `Companies` コンポーネントの認証処理を改善しました。
   - `Supabase` のモックを修正し、認証とデータ取得の処理を最適化しました。
   - `ログイン・サインアップページ` の認証フローを調整しました。

3. **テストの追加**
   - `Companies` コンポーネントのユニットテストを追加しました。
   - E2Eテストを追加しました（`e2e/login_then_logout.test.ts`）。

4. **設定ファイルの更新**
   - Jest の設定とセットアップを更新しました（`jest.config.ts`, `jest.setup.ts`）。
   - `.gitignore` を更新し、不要なファイルを無視するようにしました（例：`full_backup.sql`, `.env.test` など）。
   - TailwindCSS 設定ファイルを調整しました。

5. **依存関係の整理**
   - `@supabase/postgrest-js` をバージョン `1.17.9` に固定（mainのものに合わせて修正）。
   - `@supabase/supabase-js` のバージョンを `2.47.3` に更新（mainのものに合わせて修正）。
   - `dotenv` の型定義パッケージ（`@types/dotenv`）を追加。
   - コンフリクト解消時に発生した依存関係ファイルの更新漏れを修正。

6. **動作確認**
   - `npm run dev` によるローカル環境での動作確認を実施し、正常に動作することを確認しました。
   - 全てのテストが正常に通過しました（`npm run test`）。
   - Lint チェック（`npm run lint`）もエラーなし。

### レビューのポイントと思われる点
- コンフリクト解消部分（特に `package-lock.json` や `src/lib/database.types.ts`）が適切か。
- `Companies` コンポーネントや認証処理の修正が妥当か。
- 新規追加したテストコードのカバレッジや内容に改善点があればご指摘ください。

kazuma-naka commented on src/__tests__/Companies.test.tsx:
テストファイルの置き場なのですが
__tests__ ではなく components/
で統一して頂けると幸いです～

https://github.com/fjordllc/agent/pull/41#issuecomment-2568594710

kazuma-naka commented on src/mocks/supabase.ts:
Header.tsx のコンポネントテストで使用しますで全消しされると少し困ります～

https://github.com/fjordllc/agent/pull/41#discussion_r1906353457

kazuma-naka commented:
お疲れ様です。
CI のテストが成功していないようです！

Jyoumama committed:
Supabase関連のファイル変更: Supabase クライアントの分離に伴うコンフリクトを解消

Jyoumama committed:
Refactor: Companies.tsx を  に移動し、不要なファイルを削除

Jyoumama committed:
Companies.tsxから環境変数の記載を分離:Companies コンポーネントを  に移動

Jyoumama committed:
@supabase/ssrの使用に伴うsrc/utils/supabase/middleware.tsの変更

Jyoumama committed:
Fix: と が未使用の警告を修正

Jyoumama committed:
修正: コンフリクト解消後に package-lock.json と package.json を再生成

Jyoumama committed:
Companies.tsxとsupabaseServer.tsの修正: Next.js 15 に対応するため Supabase クライアントと認証処理を修正

- `supabase.auth.getUser()` を `supabase.auth.getSession()` に変更
- `supabaseServer.ts` を async 関数に修正し、cookies の取得を適切に処理
- 環境変数の参照方法を `?? ""` に変更し、エラーを防止

Jyoumama committed:
ESLintとPrettierの設定を更新・整理

Jyoumama committed:
Jestの設定を更新

Jyoumama committed:
Next.jsの設定を更新

Jyoumama committed:
依存関係を更新:コンフリクトの解消

Jyoumama committed:
企業一覧画面を作成

Jyoumama committed:
ログインページの修正;ShadCN/UI を使った新デザインに統合

Jyoumama committed:
Supabaseのモックデータを修正

Jyoumama committed:
不要なテストファイルを削除

Jyoumama committed:
ESLintの設定を更新し、Prettierのプラグインを追加

Jyoumama committed:
.gitignoreファイルの更新

Jyoumama committed:
ログイン・サインアップページのデザイン修正、および Next.js 設定の更新

Jyoumama committed:
TypeScript と Jest の修正、Prettier の適用

Jyoumama committed:
CIを通すためのprettierフォーマットの適用

Jyoumama committed:
Fix: Async処理の待機とコンポーネントのレンダリング方法を修正 - `mockUserLoggedIn` を非同期で待機するように修正。 - `Header` コンポーネントのレンダリング方法を `await Header()` から `render(<Header />)` に変更。 ユーザー情報を非同期で取得するようになり、テストが確実に順番通りに実行されるようになりました。

Jyoumama committed:
Fix: モックユーザーのログイン状態を非同期で設定

-  の呼び出しを非同期で行い、 を直接返さずに適切なユーザーオブジェクトを返すように修正。

これにより、モックユーザーが非同期に設定され、テストの実行が確実に同期的に行われます。

Jyoumama committed:
Fix: tailwindcss-animateプラグインの設定

-  をインポートし、プラグイン設定を修正。
-  から  に変更し、ESモジュール形式に対応。

Tailwind CSSのアニメーション機能を有効にし、設定方法をよりモジュール化しました。

Jyoumama committed:
.prettierrcの余計な制約（シングルクォート出なければならない、セミコロンがバツなど）を削除しmainに合わせる

daiiiichi committed:
Fix: ESlint, Prettier周りの設定の修正

daiiiichi committed:
Fix: Header.tsxに"use client"を追加

komagata commented:
@daiiiichi 一からPRを作った方がやりやすいのであればそちらでもOKですよ〜
